OCPBUGS#9191: Updated the Supported OIDC providers module (#60169)
openshift/openshift-docs: modules/identity-provider-oidc-supported.adoc

[id="identity-provider-oidc-supported_{context}"]
= Supported OIDC providers

> Red Hat tests and supports certain OpenID Connect (OIDC) providers with {product-title}. The following list outlines these OIDC providers:
> 
> [IMPORTANT]
> ====
> If you use an OIDC provider with {product-title} that Red Hat did not test, consider that certain limitations might exist with using this provider with {product-title}. Regardless if this provider adheres to OpenID standards, limitations could still exist.
> ====

Reviewer: We can refer to https://developer.hashicorp.com/vault/docs/platform/k8s/helm 's statement which seems better:

_The following ... are currently supported ... It may work with other ..., but those are not supported._

```suggestion
The following OpenID Connect (OIDC) providers are tested and supported with {product-title}. It may work with other OIDC providers, but those are not supported.
```

Author: Thanks. Updated.

[id="identity-provider-oidc-supported_{context}"]
= Supported OIDC providers

> OpenID Connect (OIDC) providers that comply with OpenID standards are supported with {product-title}. The following list outlines supported OpenID Connect (OIDC) providers that are tested with {product-title}:

Reviewer: I'm not sure if the use of "**supported**" in "OpenID Connect (OIDC) providers that comply with OpenID standards are **supported** with {product-title}" is good. I doubt. Per https://url.corp.redhat.com/d603b52 , only tested are supported:
```
"Tested: Where this term is used, Red Hat has explicitly tested OpenShift (for the denoted minor
version[s]) with ... Where this term is seen customers should expect the documented features
connected/related to this integration to work, and where they do not, users should open a
support case so that the issue can be investigated and addressed."
```

Author: Thanks, @xingxingxia . I updated the content. Please let me know your thoughts.

> Red Hat tests and supports certain OpenID Connect (OIDC) providers with {product-title}. The following OpenID Connect (OIDC) providers are tested and supported with {product-title}. Using an OIDC provider that is not on the following list might work with {product-title}, but the provider was not tested by Red Hat and therefore is not supported by Red Hat.

Reviewer 2: I know with support language it is kind of tricky changing things. I wonder though if changing `certain` to `specific` gets us more clarity. Certain seems a bit open for interpretation where you've got a very specific list.
